bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/owserver/OwserverConnection.java

```diff
@@ -116,24 +117,27 @@ public void stop() {
      *
      * @return a list of device ids
      */
-    public List<SensorId> getDirectory(String basePath) throws OwException {
+    public @NonNullByDefault({}) List<SensorId> getDirectory(String basePath) throws OwException {
```

Comment:
Why was `@NonNullByDefault({})` added here?

Reply:
I'd rather not use it but It has issues with captures:

`Null type mismatch (type annotations): required 'Collector<? super @Nullable SensorId,Object,@NonNull List<@NonNull SensorId>>' but this expression has type 'Collector<@NonNull SensorId,capture#of ?,List<@NonNull SensorId>>'`

Reply:
I guess it is better than casting the return value to `List<SensorId>`

Reply:
Casting also seems to work. They are both workarounds and I don't really prefer one over the other. ;-)